s3pipeline/pipelines.py

```diff
@@ -66,15 +63,15 @@ def open_spider(self, spider):
         self.ts = datetime.utcnow().replace(microsecond=0).isoformat().replace(':', '-')
         self._spider = spider
         self._timer = None
+        self._upload_chunk()
```

Comment:
Unfortunately this does not work. Since `self._timer` is None, `self._timer_cancel()` in `_upload_chunk` raises an exception `AttributeError: 'NoneType' object has no attribute 'cancel'`.

I prefer the previous implementation that cancel and restart timer on every call of  `process_item`.

Reply:
What is your test case? I cannot reproduce this error.
In the `_timer_cancel` function, it will check whether `self._timer` is not None, and then call `self._timer.cancel()`.

Reply:
Oops! I'm sorry again. The exception was raised in `close_spider` not in `_upload_chunk`. The issue was fixed in https://github.com/orangain/scrapy-s3pipeline/pull/4/commits/5239a32c50ac216dace41631d3e7f48d307c6f0a. Thanks!